src/include/wiredtiger.in

```diff
@@ -1898,6 +1906,25 @@ struct __wt_session {
 	 * @errors
 	 */
 	int __F(timestamp_transaction)(WT_SESSION *session, const char *config);
+
+	/*!
+	 * Set a timestamp on a transaction numerically.  Prefer this over @ref
+	 * timestamp_transaction when the string parsing done in that method becomes a bottleneck.
+	 *
+	 * The WT_SESSION.timestamp_transaction_uint method can only be used at snapshot
+	 * isolation.
+	 *
+	 * @requires_transaction
+	 *
+	 * @param session the session handle
+	 * @param which the timestamp you wish to set. See @ref WT_TS_TXN_TYPE for available
+	 * options, and @ref timestamp_transaction for constraints on when timestamps can be
+	 * set.
+	 * @param ts the timestamp. Zero is not allowed.
```

Comment:
Part of my slowness in reviewing is based on hesitation about disallowing 0. I think of it as a valid timestamp, and I'd prefer this API to allow setting a timestamp of 0. I've been a bit hesitant about pushing for it, since I suspect it forces some more significant changes to how this patch works.

Reply:
@agorrod, why do you want 0 to be a valid timestamp?